src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -217,13 +220,25 @@ const DomainFragment = ({
 
     if (willFollow) {
       try {
+        addIFollowByID(
+          {
+            domain_id_followed: item.domain_id_followed
+          },
+          newsDispatch
+        );
+
         await followDomain(data);
         onDomainPageFollowButtonClicked(section);
       } catch (e) {
         handleDomain(from, !willFollow, item, index);
       }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `addIFollowByID` function to add a domain to the list of followed domains is a good approach. However, error handling could be improved. In case of an error, it's better to provide more information about what went wrong. Consider logging the error or displaying a user-friendly message.
```diff
+ addIFollowByID(
+   {
+     domain_id_followed: item.domain_id_followed
+   },
+   newsDispatch
+ );
+ await followDomain(data);
+ onDomainPageFollowButtonClicked(section);
- } catch (e) {
+ } catch (error) {
+   console.error('Error following domain:', error);
+   handleDomain(from, !willFollow, item, index);
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->